analyses/cms-open-data-ttbar/ttbar_analysis_pipeline.py

```diff
@@ -182,112 +174,99 @@ def process(self, events):
             xsec_weight = 1
 
         #### systematics
-        # example of a simple flat weight variation, using the coffea nanoevents systematics feature
-        if process == "wjets":
-            events.add_systematic("scale_var", "UpDownSystematic", "weight", flat_variation)
-
         # jet energy scale / resolution systematics
         # need to adjust schema to instead use coffea add_systematic feature, especially for ServiceX
         # cannot attach pT variations to events.jet, so attach to events directly
         # and subsequently scale pT by these scale factors
-        events["pt_nominal"] = 1.0
         events["pt_scale_up"] = 1.03
         events["pt_res_up"] = jet_pt_resolution(events.Jet.pt)
 
-        pt_variations = ["pt_nominal", "pt_scale_up", "pt_res_up"] if variation == "nominal" else ["pt_nominal"]
-        for pt_var in pt_variations:
+        syst_variations = ["nominal"]
+        jet_kinematic_systs = ["pt_scale_up", "pt_res_up"]
+        event_systs = [f"btag_var_{i}" for i in range(4)]
+        if process == "wjets":
+            event_systs.append("scale_var")
+
+        # Only do systematics for nominal samples, e.g. ttbar__nominal
+        if variation == "nominal":
+            syst_variations.extend(jet_kinematic_systs)
+            syst_variations.extend(event_systs)
 
+        # for pt_var in pt_variations:
+        for syst_var in syst_variations:
             ### event selection
             # very very loosely based on https://arxiv.org/abs/2006.13076
 
-            selected_electrons = events.Electron[(events.Electron.pt > 25)] # require pt > 25 GeV for electrons
-            selected_muons = events.Muon[(events.Muon.pt > 25)] # require pt > 25 GeV for muons
-            jet_filter = (events.Jet.pt * events[pt_var]) > 25 # pT > 25 GeV for jets (scaled by systematic variations)
-            selected_jets = events.Jet[jet_filter]
-
-            # single lepton requirement
-            event_filters = ((ak.count(selected_electrons.pt, axis=1) + ak.count(selected_muons.pt, axis=1)) == 1)
-            # at least four jets
-            pt_var_modifier = events[pt_var] if "res" not in pt_var else events[pt_var][jet_filter]
-            event_filters = event_filters & (ak.count(selected_jets.pt * pt_var_modifier, axis=1) >= 4)
-            # at least one b-tagged jet ("tag" means score above threshold)
+            # Note: This creates new objects, distinct from those in the 'events' object
```

Comment:
Is this a feature of `awkward`, or how come this happens? It is not intuitive to me that the following lines create new objects.

Reply:
I agree it isn't intuitive and it's something that occurs (afaik) for the `events` object, so I think it's more of a `coffea` thing than an `awkward` thing. I'll admit that I'm a bit leery of doing this since I don't fully understand/appreciate the details myself.

Reply:
Seems like it is a cofffea thing, @nsmith- might have the details.

Reply:
Fetching a sub-array from a record array in awkward does give you a shallow copy rather than a reference. e.g. adding or modifying the fields on the shallow copy will not mutate the original:
```python
array = ak.zip({"a": ak.zip({"x": np.ones(3), "y": np.ones(3)}), "b": np.ones(3)})
sub = array.a
sub["z"] = np.ones(3)
assert("z" not in array.a.fields)
```
Of course the underlying arrays are not copied, just the "dictionary" of references

Reply:
Thanks a lot for confirming. Also very good to know that this will not copy the data around, which I had wondered about too. Then this should be behaving as we want.